components/service/glean/src/main/java/mozilla/components/service/glean/storages/ExperimentsStorageEngine.kt

```diff
@@ -0,0 +1,137 @@
+
+    data class RecordedExperimentData(
+        val branch: String,
+        val extra: Map<String, String>? = null
+    )
+
+    private val experiments: MutableMap<String, RecordedExperimentData> = mutableMapOf()
+
+    /**
+     * Record an experiment as active.
+     *
+     * @param experimentId the id of the experiment
+     * @param branch the active branch of the experiment
+     * @param extra an optional, user defined String to String map used to provide richer experiment
+     *              context if needed
+     */
+    fun setExperimentActive(
+        experimentId: String,
+        branch: String,
+        extra: Map<String, String>? = null
+    ) {
+        val truncatedExperimentId = experimentId.let {
+            if (it.length > MAX_ID_LENGTH) {
+                logger.warn("experimentId ${it.length} > $MAX_ID_LENGTH")
+                return@let it.substring(0, MAX_ID_LENGTH)
+            }
+            it
+        }
+
+        val truncatedBranch = branch.let {
+            if (it.length > MAX_ID_LENGTH) {
+                logger.warn("branch ${it.length} > $MAX_ID_LENGTH")
+                return@let it.substring(0, MAX_ID_LENGTH)
+            }
+            it
+        }
+
+        val experimentData = RecordedExperimentData(truncatedBranch, extra)
+        experiments.put(truncatedExperimentId, experimentData)
+    }
+
+    /**
+     * Set an active experiment as inactive.
+     *
+     * Warns to the logger (but does not raise) if the experiment is not already active.
+     *
+     * @param experimentId the id of the experiment to set as inactive
+     */
+    fun setExperimentInactive(experimentId: String) {
+        if (!experiments.containsKey(experimentId)) {
+            logger.warn("experiment $experimentId} not already active")
+        } else {
+            experiments.remove(experimentId)
+        }
+    }
+
+    /**
+     * Retrieves the data about active experiments
+     *
+     * @return the map of experiment ids to [RecordedExperimentData]
+     */
+    @Synchronized
+    fun getSnapshot(): Map<String, RecordedExperimentData> {
+        return experiments
+    }
+
+    private fun convertExtra(extra: Map<String, String>): JSONObject {
+        val jsonExtra = JSONObject()
+        for ((k, v) in extra) {
+            jsonExtra.put(k, v)
+        }
+        return jsonExtra
+    }
+
+    /**
+     * Get a snapshot of the stored data as a JSON object.
+     *
+     * @param storeName the name of the desired store.  Since experiments are
+     *   included in all stores, this parameter is effectively ignored, but
+     *   must be provided to comply with the StorageEngine interface.
+     * @param clearStore whether or not to clearStore the requested store.
+     *   This parameter is ignored.
+     *
+     * @return the JSONArray containing information about the active experiments
+     */
+    override fun getSnapshotAsJSON(storeName: String, clearStore: Boolean): Any? {
```

Comment:
Let's also have test coverage for this :) You can explicitly mark `storeName` as unused by prefixing it with ` @Suppress("UNUSED_PARAMETER")`, so the signature would become 

`override fun getSnapshotAsJSON(@Suppress("UNUSED_PARAMETER") storeName: String, ...`